### ISSUE TITLE
Error (GH API) when creating full-release with CLI

### ISSUE THREAD

**Issue by sdaschner** [jreleaser/jreleaser#1191]
There seems to be an issue when attempting to create a full release via the CLI:

The reproducer is under https://github.com/sdaschner/capitalize-titles

```
# project is https://github.com/sdaschner/capitalize-titles/tree/main
$> uname -a
Linux squarelinux 6.1.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 18 Jan 2023 19:54:38 +0000 x86_64 GNU/Linux

$> java -version
openjdk version "18.0.1" 2022-04-19
OpenJDK Runtime Environment Temurin-18.0.1+10 (build 18.0.1+10)
OpenJDK 64-Bit Server VM Temurin-18.0.1+10 (build 18.0.1+10, mixed mode, sharing)

$> jreleaser --version 
------------------------------------------------------------
jreleaser 1.4.0
------------------------------------------------------------
Build time:   2022-12-29 15:08:38Z
Revision:     b977b66baed1e4761af054c5d747a0804f457252
JVM:          17.0.5 (Oracle Corporation GraalVM 22.3.0 Java 17 CE)
------------------------------------------------------------


$> mvn -P native package
$> mvn -P dist package
$> jreleaser full-release -d linux-x86_64
```

produces the following error:

```
[INFO]  Storing changelog: out/jreleaser/release/CHANGELOG.md
[DEBUG] ## Changelog

d70328a added JReleaser
5f245e5 Create LICENSE
c436155 updated ignored words
19b0888 removed trailing newline
9385f6b updated artf name
cf07b69 added picocli & native build
fa591d3 capitalize multi-line strings
175a679 updated readme
998fb0b initial commit
[INFO]  Calculating checksums
[INFO]    [checksum] No files configured for checksum. Skipping
[INFO]  Signing files
[INFO]    [sign] Signing is not enabled. Skipping
[INFO]  Deploying is not enabled. Skipping
[INFO]  Uploading is not enabled. Skipping
[INFO]  Releasing to https://github.com/sdaschner/capitalize-titles
[DEBUG] fetching all branches on sdaschner/capitalize-titles
org.kohsuke.github.HttpException: Server returned HTTP response code: 200, message: 'null' for URL: https://api.github.com/repos/sdaschner/capitalize-titles/branches
	at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:488)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:407)
	at org.kohsuke.github.GitHubPageIterator.fetch(GitHubPageIterator.java:144)
	at org.kohsuke.github.GitHubPageIterator.hasNext(GitHubPageIterator.java:91)
	at org.kohsuke.github.PagedIterator.fetch(PagedIterator.java:106)
	at org.kohsuke.github.PagedIterator.nextPageArray(PagedIterator.java:134)
	at org.kohsuke.github.PagedIterable.toArray(PagedIterable.java:78)
	at org.kohsuke.github.PagedIterable.toArray(PagedIterable.java:106)
	at org.kohsuke.github.GHRepository.getBranches(GHRepository.java:2391)
	at org.jreleaser.sdk.github.Github.listBranches(Github.java:106)
	at org.jreleaser.sdk.github.GithubReleaser.createRelease(GithubReleaser.java:181)
	at org.jreleaser.model.spi.release.AbstractReleaser.release(AbstractReleaser.java:51)
	at org.jreleaser.engine.release.Releasers.release(Releasers.java:52)
	at org.jreleaser.workflow.ReleaseWorkflowItem.doInvoke(ReleaseWorkflowItem.java:39)
	at org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:43)
	at org.jreleaser.engine.hooks.CommandHookExecutor.execute(CommandHookExecutor.java:52)
	at org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:43)
	at org.jreleaser.workflow.WorkflowImpl.doExecute(WorkflowImpl.java:111)
	at org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:51)
	at org.jreleaser.cli.FullRelease.doExecute(FullRelease.java:180)
	at org.jreleaser.cli.AbstractModelCommand.execute(AbstractModelCommand.java:92)
	at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:37)
	at org.jreleaser.cli.AbstractModelCommand.call(AbstractModelCommand.java:53)
	at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:29)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at org.jreleaser.cli.Main.execute(Main.java:73)
	at org.jreleaser.cli.Main.run(Main.java:58)
	at org.jreleaser.cli.Main.main(Main.java:49)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.kohsuke.github.GHBranch`: cannot deserialize from Object value (no delegate- or property-based Creator): this appears to be a native image, in which case you may need to configure reflection for the class that is to be deserialized
 at [Source: (String)"[{"name":"main","commit":{"sha":"5f245e5031ffa5972e0ac8890dd06a9c37ddc82f","url":"https://api.github.com/repos/sdaschner/capitalize-titles/commits/5f245e5031ffa5972e0ac8890dd06a9c37ddc82f"},"protected":false,"protection":{"enabled":false,"required_status_checks":{"enforcement_level":"off","contexts":[],"checks":[]}},"protection_url":"https://api.github.com/repos/sdaschner/capitalize-titles/branches/main/protection"}]"; line: 1, column: 3] (through reference chain: java.lang.Object[][0])
	at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1909)
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:408)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1349)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1417)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:352)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:185)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:216)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:26)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2105)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1546)
	at org.kohsuke.github.GitHubResponse.parseBody(GitHubResponse.java:97)
	at org.kohsuke.github.GitHubPageIterator.lambda$fetch$0(GitHubPageIterator.java:145)
	at org.kohsuke.github.GitHubClient.createResponse(GitHubClient.java:449)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:399)
	... 33 more
[INFO]  Writing output properties to out/jreleaser/output.properties
[ERROR] JReleaser failed after 0.650 s
org.jreleaser.model.JReleaserException: Unexpected error when creating release
	at org.jreleaser.workflow.ReleaseWorkflowItem.doInvoke(ReleaseWorkflowItem.java:41)
	at org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:43)
	at org.jreleaser.engine.hooks.CommandHookExecutor.execute(CommandHookExecutor.java:52)
	at org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:43)
	at org.jreleaser.workflow.WorkflowImpl.doExecute(WorkflowImpl.java:111)
	at org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:51)
	at org.jreleaser.cli.FullRelease.doExecute(FullRelease.java:180)
	at org.jreleaser.cli.AbstractModelCommand.execute(AbstractModelCommand.java:92)
	at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:37)
	at org.jreleaser.cli.AbstractModelCommand.call(AbstractModelCommand.java:53)
	at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:29)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at org.jreleaser.cli.Main.execute(Main.java:73)
	at org.jreleaser.cli.Main.run(Main.java:58)
	at org.jreleaser.cli.Main.main(Main.java:49)
Caused by: org.jreleaser.model.spi.release.ReleaseException: org.kohsuke.github.HttpException: Server returned HTTP response code: 200, message: 'null' for URL: https://api.github.com/repos/sdaschner/capitalize-titles/branches
	at org.jreleaser.sdk.github.GithubReleaser.createRelease(GithubReleaser.java:243)
	at org.jreleaser.model.spi.release.AbstractReleaser.release(AbstractReleaser.java:51)
	at org.jreleaser.engine.release.Releasers.release(Releasers.java:52)
	at org.jreleaser.workflow.ReleaseWorkflowItem.doInvoke(ReleaseWorkflowItem.java:39)
	... 21 more
Caused by: org.kohsuke.github.HttpException: Server returned HTTP response code: 200, message: 'null' for URL: https://api.github.com/repos/sdaschner/capitalize-titles/branches
	at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:488)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:407)
	at org.kohsuke.github.GitHubPageIterator.fetch(GitHubPageIterator.java:144)
	at org.kohsuke.github.GitHubPageIterator.hasNext(GitHubPageIterator.java:91)
	at org.kohsuke.github.PagedIterator.fetch(PagedIterator.java:106)
	at org.kohsuke.github.PagedIterator.nextPageArray(PagedIterator.java:134)
	at org.kohsuke.github.PagedIterable.toArray(PagedIterable.java:78)
	at org.kohsuke.github.PagedIterable.toArray(PagedIterable.java:106)
	at org.kohsuke.github.GHRepository.getBranches(GHRepository.java:2391)
	at org.jreleaser.sdk.github.Github.listBranches(Github.java:106)
	at org.jreleaser.sdk.github.GithubReleaser.createRelease(GithubReleaser.java:181)
	... 24 more
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.kohsuke.github.GHBranch`: cannot deserialize from Object value (no delegate- or property-based Creator): this appears to be a native image, in which case you may need to configure reflection for the class that is to be deserialized
 at [Source: (String)"[{"name":"main","commit":{"sha":"5f245e5031ffa5972e0ac8890dd06a9c37ddc82f","url":"https://api.github.com/repos/sdaschner/capitalize-titles/commits/5f245e5031ffa5972e0ac8890dd06a9c37ddc82f"},"protected":false,"protection":{"enabled":false,"required_status_checks":{"enforcement_level":"off","contexts":[],"checks":[]}},"protection_url":"https://api.github.com/repos/sdaschner/capitalize-titles/branches/main/protection"}]"; line: 1, column: 3] (through reference chain: java.lang.Object[][0])
	at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1909)
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:408)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1349)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1417)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:352)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:185)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:216)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:26)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2105)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1546)
	at org.kohsuke.github.GitHubResponse.parseBody(GitHubResponse.java:97)
	at org.kohsuke.github.GitHubPageIterator.lambda$fetch$0(GitHubPageIterator.java:145)
	at org.kohsuke.github.GitHubClient.createResponse(GitHubClient.java:449)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:399)
	... 33 more
```

**Comment by aalmiray** [comment]
The native executable generated by JReleaser is missing reflection information from hub4j.

**Comment by sdaschner** [comment]
> The native executable generated by JReleaser

What does that mean? My understanding was, the `mvn` commands create the executables (under `target/distributions` as described in the assembly, and that JReleaser creates the release using these files? I used the standard approach to create a native Quarkus CLI, which perfectly works...

**Comment by aalmiray** [comment]
I meant the `jreleaser` native executable you used for releasing.

**Comment by sdaschner** [comment]
That fixed it, tried the new snapshot version, thanks!

**Comment by aalmiray** [comment]
Released in v1.5.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.5.0
